server/client.go

```diff
@@ -1010,8 +1010,18 @@ func (c *client) traceMsg(msg []byte) {
 	if !c.trace {
 		return
 	}
-	// FIXME(dlc), allow limits to printable payload.
-	c.Tracef("<<- MSG_PAYLOAD: [%q]", msg[:len(msg)-LEN_CR_LF])
+
+	origLength := len(msg) - LEN_CR_LF
+	var msgToWrite []byte
+	if origLength > c.srv.opts.MaxTracedMessagePrintableLen {
+		msgToWrite = make([]byte, c.srv.opts.MaxTracedMessagePrintableLen+len("..."))
```

Comment:
We should avoid allocation here IMO.

Reply:
Otherwise, we need to trick the log formatter.

Reply:
I am sure its possible and prefer a solution that does not do additional allocs if possible.